src/dbnode/storage/types.go

```diff
@@ -282,7 +291,15 @@ type databaseNamespace interface {
 		ctx context.Context,
 		query index.Query,
 		opts index.QueryOptions,
-	) (index.QueryResults, error)
+	) (index.QueryReturnResults, error)
+
+	// AggregateQuery resolves the given query into aggregated tags.
+	AggregateQuery(
+		ctx context.Context,
+		query index.Query,
+		opts index.QueryOptions,
+		aggResultOpts *index.AggregateResultsOptions,
```

Comment:
Since this method is already a `AggregateQuery(...)` call, you definitely need to pass `aggResultOpts` so can we remove the pointer type on this?